accounts-db/src/bucket_map_holder_stats.rs

```diff
@@ -253,29 +254,31 @@ impl BucketMapHolderStats {
                     ),
                 );
             }
+            let count_in_mem = self.count_in_mem.load(Ordering::Relaxed);
+            let held_in_mem_ref_count = self.held_in_mem.ref_count.swap(0, Ordering::Relaxed);
+            let held_in_mem_slot_list_len =
+                self.held_in_mem.slot_list_len.swap(0, Ordering::Relaxed);
+            // Assume if an entry is held in-mem that is has (at least) two slot list entries.
+            // Since `approx_size_of_one_entry()` already includes the ref count & metadata sizes,
+            // only add in a second slot list entry.
+            let estimate_mem_bytes = count_in_mem
+                * InMemAccountsIndex::<T, U>::approx_size_of_one_entry()
+                + (held_in_mem_ref_count + held_in_mem_slot_list_len) as usize
```

Comment:
I think `ref_count > 1` doesn't necessarily implies that slot_list>=2.  slot_list could be 1 and ref_count > 1.

Reply:
I remember seeing that account index entry with slot_list.len = 1 and ref_count > 1. 
afaik, In clean, if we see multiple elements in slot_list, we keep the last item and remove the old items from slot_list. but ref_count stays the same in `clean`. ref_count is updated in shrink. 

> And not counting during clean/shrink, then there must be at least ref_count entries in the slot list too.
I think not every cleaned items from the index is processed by the shrink. Some of them may not be shrunk.
immediately. 

I think we can validate this assumption by adding an assert ,where we update ref_count to check that slot_list_len == ref_count?